[opentracing] v1.5.1

david-antiteum opened the pull request:
This is the first commit of an OpenTracing port.

Issue: #5970

david-aimsun committed:
OpenTracing #5970

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5974) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5974) before we can accept your contribution.<br/><br/>:x: david-antiteum [sign now](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5974)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vcpkg?pullRequest=5974) it.</sub>

Rastaban commented:
Thanks for the new port!  It looks like the build is failing on the UWP ports and x64-windows-static.  Are these failures expected?  If so we can probably get this checked in.

david-antiteum commented:
> Thanks for the new port! It looks like the build is failing on the UWP ports and x64-windows-static. Are these failures expected? If so we can probably get this checked in.

The static version can only be build with the dynamic version (a limitation in the project cmake). The dynamic version can be build without the static. Is this OK?

```
if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
	set( LOCAL_OPTIONS
		-DBUILD_STATIC_LIBS=OFF
	)
else()
    message("Static building is only possible when compiling static and dynamic versions at the same time. Enabling both.")
    set(VCPKG_LIBRARY_LINKAGE dynamic)
endif()
```

This will generate, for the static build:

```
find_package(OpenTracing CONFIG REQUIRED)
target_link_libraries(main PRIVATE OpenTracing::opentracing OpenTracing::opentracing-static OpenTracing::opentracing_mocktracer OpenTracing::opentracing_mocktracer-static)
```

And, for the dynamic build:

```
find_package(OpenTracing CONFIG REQUIRED)
target_link_libraries(main PRIVATE OpenTracing::opentracing OpenTracing::opentracing_mocktracer)
```

Then, for UWP. Is this enough?

```
if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL WindowsStore)
    message(FATAL_ERROR "Error: UWP build is not supported.")
endif()
```

Thanks

david-aimsun committed:
No UWP. Static requires dynamic #5970
- Dynamic version can be build without the static one
- Static version can only be build when dynamic is also active
- UWP is not supported

david-antiteum committed:
Linux support #5970

Rastaban commented:
I am happy with those changes, thank you for adding messages to make it clear what it is doing.
